modin/pandas/__init__.py

```diff
@@ -111,15 +111,23 @@ def _update_engine(publisher: Parameter):
             from modin.core.execution.ray.common.utils import initialize_ray
 
             initialize_ray()
-    elif publisher.get() == "Native":
+    elif publisher.get() == "Native" and StorageFormat.get() not in [
```

Comment:
This is not what I originally expected. While this might work, this is doing it in a wrong way.

You should just check that `StorageFormat` is not explicitly set by user by doing something like `StorageFormat.get_value_source() == ValueSource.DEFAULT`, and then set it.

Same goes to user setting `StorageFormat` to `Omnisci` but not setting `Engine` to `Native`.